File: models/Project.go
package models

import (
	"time"
)

// Project is the top level structure in Semaphore
type Project struct {
	ID        int       `db:"id" json:"id"`
	Name      string    `db:"name" json:"name" binding:"required"`
	Created   time.Time `db:"created" json:"created"`
	Alert     bool      `db:"alert" json:"alert"`
	AlertChat string    `db:"alert_chat" json:"alert_chat"`
}

File: public/js/factories/project.js
app.factory('ProjectFactory', ['$http', function ($http) {
	var Project = function (project) {
		this.id = project.id;
		this.name = project.name;
		this.alert = project.alert;
		this.alert_chat = project.alert_chat;
	}

	Project.prototype.getURL = function () {
		return '/project/' + this.id;
	}

	return Project;
}]);